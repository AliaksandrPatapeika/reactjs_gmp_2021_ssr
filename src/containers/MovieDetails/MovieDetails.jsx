import React from 'react';
import styled from 'styled-components';

import Blur from '../../components/Blur';
import MovieDetailsContent from '../../components/MovieDetailsContent';
import HeaderTop from '../HeaderTop/HeaderTop';

const MovieDetailsWrapper = styled.div`
  margin-bottom: 10px;
  background-image: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)), url('/img/bgImage.jpg');
  background-size: cover;
`;

const MovieDetails = () => (
  <MovieDetailsWrapper>
    <Blur>
      <HeaderTop />
      <MovieDetailsContent />
    </Blur>
  </MovieDetailsWrapper>
);

export default MovieDetails;
