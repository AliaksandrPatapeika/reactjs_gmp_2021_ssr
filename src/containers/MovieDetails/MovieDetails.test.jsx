import {head} from 'lodash';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../tests/mocks/mockData';
import MovieDetails from './MovieDetails';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {id: '12'}
  })
}));

describe('MovieDetails', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        activeMovieDetailsMovie: mockMovie
      }
    };
    const store = mockStore(initialState);
    const movieDetails = renderer
      .create(
        <Provider store={store}>
          <MovieDetails title="title" />
        </Provider>
      )
      .toJSON();

    expect(movieDetails).toMatchSnapshot();
  });
});
