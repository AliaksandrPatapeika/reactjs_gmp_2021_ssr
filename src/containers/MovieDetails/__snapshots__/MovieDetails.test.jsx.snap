// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div
  className="MovieDetails__MovieDetailsWrapper-sc-5pmrvb-0 cRZVRB"
>
  <div
    className="Blur__BlurWrapper-b9kkbo-0 jwekOJ"
  >
    <div
      className="headerTopContainer"
    >
      <div
        className="logoContainer"
      >
        <span
          className="netflix"
        >
          netflix
        </span>
        <span
          className="roulette"
        >
          roulette
        </span>
      </div>
      <button
        className="returnToMainButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <i
          aria-hidden="true"
          className="fa fa-search fa-flip-horizontal"
        />
      </button>
    </div>
    <div
      className="movieDetailsContentContainer"
    >
      <img
        alt="movie details poster"
        className="movieDetailsPoster"
        onError={[Function]}
        src="http://localhost:3000/img/01-pulp-fiction.jpg"
      />
      <div
        className="movieDetailsContent"
      >
        <div
          className="movieDetailsTitleRating"
        >
          <span
            className="movieDetailsTitle"
          >
            Pulp Fiction
          </span>
          <div
            className="movieDetailsRating"
            title="Vote count: 1195"
          >
            4.3
          </div>
        </div>
        <div>
          Mock tagline!
        </div>
        <div
          className="movieDetailsReleaseDateRuntime"
        >
          <div
            title="Release date: 2004-03-15"
          >
            2004
          </div>
          <div>
            154
             
            min
          </div>
        </div>
        <div
          className="movieDetailsLabel"
        >
          Genres
          : 
          <span
            className="movieDetailsValue"
          >
            Action & Adventure
          </span>
        </div>
        <div
          className="movieDetailsLabel"
        >
          Budget
          : 
          <span
            className="movieDetailsValue"
          >
            55 000 000 $
          </span>
        </div>
        <div
          className="movieDetailsLabel"
        >
          Revenue
          : 
          <span
            className="movieDetailsValue"
          >
            136 906 000 $
          </span>
        </div>
        <div
          className="overview"
        >
          Pulp Fiction is a 1994 American neo-noir black comedy crime film written and directed by Quentin Tarantino, who conceived it with Roger Avary.[4] Starring John Travolta, Samuel L. Jackson, Bruce Willis, Tim Roth, Ving Rhames, and Uma Thurman, it tells several stories of criminal Los Angeles. The title refers to the pulp magazines and hardboiled crime novels popular during the mid-20th century, known for their graphic violence and punchy dialogue.
        </div>
      </div>
    </div>
  </div>
</div>
`;
